File: src/main/java/com/googongill/aditory/security/jwt/user/PrincipalDetails.java
package com.googongill.aditory.security.jwt.user;

import com.googongill.aditory.domain.User;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.ArrayList;
import java.util.Collection;

public class PrincipalDetails implements UserDetails {
    private User user;

    public PrincipalDetails(User user) {
        this.user = user;
    }

    // 해당 유저 권한 반환
    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        ArrayList<GrantedAuthority> collect = new ArrayList<>();
        collect.add((GrantedAuthority) () -> String.valueOf(user.getRole()));
        return collect;
    }

    public Long getUserId() {
        return user.getId();
    }

    public String getNickname() {
        return user.getNickname();
    }

    @Override
    public String getUsername() {
        return user.getUsername();
    }

    @Override
    public String getPassword() {
        return user.getPassword();
    }

    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

    @Override
    public boolean isAccountNonLocked() {
        return true;
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public boolean isEnabled() {
        return true;
    }
}

File: src/main/java/com/googongill/aditory/security/jwt/user/PrincipalDetails.kt
package com.googongill.aditory.security.jwt.user

import com.googongill.aditory.domain.User
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.UserDetails

class PrincipalDetails(val user: User) : UserDetails {
    // 해당 유저 권한 반환
    override fun getAuthorities(): Collection<GrantedAuthority> {
        val collect = ArrayList<GrantedAuthority>()
        collect.add(GrantedAuthority { user.role.toString() })
        return collect
    }

    val userId: Long?
        get() = user.id
    val nickname: String?
        get() = user.nickname

    override fun getUsername(): String {
        return user.username
    }

    override fun getPassword(): String {
        return user.password
    }

    override fun isAccountNonExpired(): Boolean {
        return true
    }

    override fun isAccountNonLocked(): Boolean {
        return true
    }

    override fun isCredentialsNonExpired(): Boolean {
        return true
    }

    override fun isEnabled(): Boolean {
        return true
    }
}
